src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,35 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(country);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't save country to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return country;
     }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        Country country;
+        try (Session session = factory.openSession()) {
+            country = session.get(Country.class, id);
+            session.close();
```

Comment:
You don't need to call `session.close()` while using try-with-resources

Reply:
agree with @viacheslav-torbin